### PR TITLE
Adding AI & ML topics to MCW

### PR DESCRIPTION
**Pull request by benstegink**

### CODE REVIEW

**Comment by DawnmarieDesJardins** [comment]
After review and feedback, this workshop will be renamed and published as Azure Synapse Analytics and AI

**Comment by dem108** [comment]
Commit bd3569abf3f96d75426cb2b5f76b7390f51f1965 has changed ML Exercise from 6 to 7, but the content still refers to Exercise 6. For example, Task 1: Training models starts by opening the **ASAMCW - Exercise 6 - Machine Learning** notebook, which should be changed to **Exercise 7**. The corresponding name of the notebook should also be changed.

File: Hands-on lab/HOL step-by step - Azure Synapse Analytics end-to-end solution.md [hunk 1]
```diff
@@ -49,15 +49,20 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
   - [Exercise 4: Exploring raw text based data with Azure Synapse SQL Serverless](#exercise-4-exploring-raw-text-based-data-with-azure-synapse-sql-serverless)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Azure Synapse Analytics end-to-end solution.md:49]: Adding review comment as a test.

**Comment by DawnmarieDesJardins** [comment]
Feedback via email:
This is a very good hands-on lab. I implemented it step by step and introduced it to the team in the Azure Synapse-AI session in GBB Learning Week.

There's only one problem.  We discussed it during the GBB learning week about the inability of SQL prediction functions to execute.
in Exercise 6: Machine Learning - Task 3: Making predictions with the registered model.
5. Create another new SQL script and replace the contents with the following:
-- Retrieve the latest hex encoded ONNX model from the table
DECLARE @model varbinary(max) = (SELECT Model FROM [wwi_mcw].[ASAMCWMLModel] WHERE Id = (SELECT Top(1) max(ID) FROM [wwi_mcw].[ASAMCWMLModel]));

-- Run a prediction query
SELECT d.*, p.*
FROM PREDICT(MODEL = @model, DATA = [wwi_mcw].[ProductPCA] AS d) WITH (prediction real) AS p;

Please highlight in the MCW the fact that this is not working yet.

**Comment by DawnmarieDesJardins** [comment]
Feedback via email:
Some suggestion here:

Suggestion 1:
In Exercise 6: Machine Learning
We know that all steps in the PowerShell script have been created with an automated script in session of Before the HOL in ./01-environment-setup.ps1.

Please help to add instructions on how to connect Azure Synapse and Data Lake using AML step by step.
We need to make sure that the participants also know how to configure app registrations to get client and tenant IDs to connect to the data lake using Azure Machine Learning.

**Comment by nchandhi** [comment]
Please add Auto ML model registration to AML and deployment to ACI and AKS as discussed in our call on 07/16.

**Comment by timahenning** [comment]
> Please add Auto ML model registration to AML and deployment to ACI and AKS as discussed in our call on 07/16.

Spoke to Zoiner.  We will incorporate your suggested changes in time for James Serra's demo on 7/21.  We use ACI because this is a development environment.

**Comment by DawnmarieDesJardins** [comment]
SME review for this update is closed.  Changes and feedback above are being incorporated and will be merged soon.

File: Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md [hunk 2]
```diff
@@ -1001,19 +996,536 @@ A common format for exporting and storing data is with text based files. These c
...
+2. Select **Create**.
+
+    ![The Cognitive Services overview form is displayed with the Create button highlighted.](media/ex5-task2a-02.png "The Cognitive Services overview form")
+
+3. Enter the following configuration settings, then select **Create**:
+
+    | Field | Value |
+    |-------|-------|
+    | Name  | Enter a unique name (denoted by the green checkmark indicator) for the form recognition service. |
+    | Subscription | Select the lab subcription. |
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md:1115]: Typo - subscription

File: Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md [hunk 3]
```diff
@@ -1001,19 +996,536 @@ A common format for exporting and storing data is with text based files. These c
...
+
+17. Name your pipeline **InvoiceProcessing**.
+
+    ![The new pipeline properties are shown with InvoiceProcessing entered as the name of the pipeline.](media/ex5-task4-014.png "Naming the pipeline")
+
+18. On the pipeline taskbar, select **Add trigger** then choose **New/Edit** to create an event to start the pipeline.
+
+    ![The Add trigger button is expanded with the New/Edit option selected.](media/ex5-task4-015.png "New Trigger menu item")
+
+19. On the Add triggers form, select  **+New** from the the **Choose trigger** dropdown.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md:1365]: typo - the the

File: Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md [hunk 4]
```diff
@@ -49,15 +49,18 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
   - [Exercise 4: Exploring raw text based data with Azure Synapse SQL Serverless](#exercise-4-exploring-raw-text-based-data-with-azure-synapse-sql-serverless)
     - [Task 1: Query CSV data](#task-1-query-csv-data)
     - [Task 2: Query JSON data](#task-2-query-json-data)
-  - [Exercise 5: Security](#exercise-5-security)
+  - [Exercise 5: Synapse Pipelines & Cognitive Search](#exercise-5-synapse-pipelines--cognitive-search)
+    - [Task 1: Create the necessary storage account](#task-1-create-the-necessary-storage-account)
+    - [Task 2: Create and train Azure Forms recognizer model and Cognitive Search](#task-2-create-and-train-azure-forms-recognizer-model-and-cognitive-search)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md:54]: Is recognizer part of Azure Forms?  I think it should be Azure Forms Recognizer (capital R)

File: Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md [hunk 5]
```diff
@@ -49,15 +49,18 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
...
     - [Task 3: Dynamic data masking](#task-3-dynamic-data-masking)
-  - [Exercise 6: Machine Learning](#exercise-6-machine-learning)
-    - [Task 1: Training models](#task-1-training-models)
-    - [Task 2: Registering the models with Azure Synapse Analytics](#task-2-registering-the-models-with-azure-synapse-analytics)
-    - [Task 3: Making predictions with the registered models](#task-3-making-predictions-with-the-registered-models)
-  - [Exercise 7: Monitoring](#exercise-7-monitoring)
+  - [Exercise 7: Machine Learning](#exercise-7-machine-learning)
+    - [Task 1: Training, consuming, and deploying models](#task-1-training-consuming-and-deploying-models)
+  - [Exercise 8: Monitoring](#exercise-8-monitoring)
     - [Task 1: Workload Importance](#task-1-workload-importance)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by step - Azure Synapse Analytics and AI.md:64]: Tasks 1 & 2 - Should be Workload importance and Workload isolation (small i) they aren't products.

File: Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md [hunk 6]
```diff
@@ -549,6 +555,18 @@ The primary audience is the business decision makers and technology decision mak
 
     The data team can then launch notebooks attached to the Azure Synapse Spark pool and author the code that uses their favorite libraries.  
 
+6. How does your solution address their need to keep their part costs table in the data warehouse updated by the supplier invoices? 
+
+    WWI can accomplish this by a combination of a Synapse Pipeline with an Azure Cogntive Search Skillset that invokes the Form Recognizer service as a custom skill. The pipeline would work as follows:
```
Comment (DawnmarieDesJardins) [inline Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md:560]: Typo - Cognitive

File: Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md [hunk 7]
```diff
@@ -549,6 +555,18 @@ The primary audience is the business decision makers and technology decision mak
...
+
+     - Invoice is uploaded to Azure Storage.
+     - This triggers a Synapse Pipeline.
+     - The Synapse Pipeline has a web activity that invokes an Azure Cognitive Search skillset.
+     - The first skill in the skillset invokes an Azure Function, passing it the URL to the PDF invoice. 
+     - The Azure Function invokes the Form Recognizer service, passing it the URL and SAS token to the PDF invoice. Forms recognizer returns the OCR results to the function.
+     - The Azure Function returns the results to skillset. The skillset then extracts only the product names and costs and sends that to a configure knowledge store that writes the extracted data to JSON files in Azure Storage.
+     - The Synapse pipline reads these JSON files from Azure Storage in a Data Flow activity and performs an upsert against the product catalog table in the Synapse SQL Pool.
```
Comment (DawnmarieDesJardins) [inline Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md:568]: Typo - pipeline

**Comment by DawnmarieDesJardins** [comment]
Student/Trainer Guides - 

Design
Under Transform - the trainer guide answers 6 questions, in the student guides, only 5 are asked.
Missing - 
6. How does your solution address their need to keep their part costs table in the data warehouse updated by the supplier invoices?

Customer Objections - 
Student guides give 7 customer objections - trainer guide answers 9 objections.
Missing - 
8. How does Azure support deploying the models as web services so that they can easily be invoked from client applications? How does a model get deployed as a webservice?
9. What does the model re-training process look like in Azure? How can WWI data scientists train and evaluate new models but also ensure this re-training becomes part of the DevOps process used to deploy any updates to the application. Can Azure help them orchestrate updates to the client applications, the machine learning API's and the models that power the API's?

**Comment by DawnmarieDesJardins** [comment]
Exercise 5, task 1, instruction 11
Coding jumps from bullets to i. ii. iii.  please correct.

File: Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md (hunk 7 above)
Comment (codingbandit) [reply Whiteboard design session/WDS trainer guide - Azure Synapse Analytics and AI.md:568]: Yep, that's a thing :)

**Pull request merged**